draft-ietf-ppm-dap.md

```diff
@@ -484,6 +480,132 @@ struct {
+is encoded by the "query configuration" configured out-of-band:
+
+~~~
+struct {
+    QueryType query_type;
+    Duration time_precision;
+    uint64 min_batch_size;
+    select (query_type) {
+        case time-interval:
+            Empty;
+        case fixed-size:
+            uint64 max_batch_size;
+    }
+} QueryConfig;
+~~~
+
+### Time-interval Queries
+
+The first query type, `time-interval`, is designed to support applications in
+which some process is observed over a long period of time. The Collector
+specifies a "batch interval" that determines the time range for measurements
+included in the batch: For each measurement in the batch, the time at which that
+measurement was generated (see {{upload-flow}}) must fall within the batch
+interval specified by the Collector.
+
+The configuration of this query type specifies two paramters:
+
+- `min_batch_size` - Is the smallest number of measurements the batch is allowed
+  to include. In a sense, this parameter controls the degree of privacy that
+  will be obtained: The larger the minimum batch size, the higher degree of
+  privacy. However, this ultimately depends on the application and the nature of
+  the measurements and aggregation function.
+
+- `time_precision` - The duration of the smallest permitted batch interval.
+  Additional restrictions apply; see {{batch-validation}}. In addition, Clients
+  use this value to truncate their report timestamps; see {{upload-flow}}.
+
+Typically the Collector issues queries for which the batch intervals are
+continuous, monotonically increasing, and have the same duration. For example,
+the sequence of batch intervals. For example, the sequence of batch intervals
+`(1659544000, 1000)`, `(1659545000, 1000)`, `(1659545000, 1000)`, `(1659546000,
+1000)` satisfies these conditions. (The first element of the pair denotes the
+start of the batch interval and the second denotes the duration.) Of course,
+there are cases in which Collector may need to issue queries out-of-order. For
+example, a previous batch might need to be queried again with a different
+aggregation parameter (e.g, for Poplar1). In addition, the Collector may need to
+vary the duration to adjust to changing measurement upload rates.
+
+### Fixed-size Queries
+
+The `fixed-size` query type is used to support applications in which the
+Collector needs the ability to strictly control the sample size. This is
+particularly important for controlling the amount of noise added to measurements
+by Clients (or added to aggregate shares by Aggregators) in order to achieve
+differential privacy.
+
+For this query type, the Aggregators batch measurements into arbitrary batches
+such that each batch has roughly the same number of reports. The configuration
+includes parameters `min_batch_size` and `max_batch_size` that determine,
+respectively, the minimum and maximum number of measurements per batch.
+
+To get the aggregate of the next batch, the Collector issues a query of type
```

Comment:
Do we want to describe the behaviour when collector calls with `next-batch` before the last `batch-id` has been returned? I'm not sure there is a use case for collecting fixed sized batches in parallel. One of the reasons to have fixed size collection is to close a batch asap so any iterative tasks can move on to the next iteration, if there are two opening `batch-id`s for one task, then I imagine the aggregators would distribute reports to these two batches and make the collection slower for one batch? @cjpatton

Reply:
> One of the reasons to have fixed size collection is to close a batch asap so any iterative tasks can move on to the next iteration, ...

Is there anything in the spec that precludes this? 

> if there are two opening `batch-id`s for one task, then I imagine the aggregators would distribute reports to these two batches and make the collection slower for one batch?

It seems like this is a matter for the implementation. I agree this behavior is permitted, but the Leader could also fill the first batch first, then the second.